code/modules/food_and_drinks/drinks/drinks/cans.dm

```diff
@@ -29,7 +57,16 @@
 	return crushed_can
 
 /obj/item/reagent_containers/food/drinks/cans/attack(mob/M, mob/user, proximity)
-	if(canopened == 0)
+	if(user.a_intent == INTENT_HARM && !canopened)
+		visible_message("<span class='warning'>[user.name] shakes up the [name]!</span>")
+		if(shaken < 5)
+			if(!shaken)
+				addtimer(CALLBACK(src, .proc/resetshake), 1 MINUTES)
```

Comment:
shaking it once then waiting 50 seconds and shaking it 3 more times will still reset after the initial 1 minute timer.
Just making sure that is intended

Reply:
I learned adding many timers isnt as bad as I thought it might be, so I changed this behavior to loose shakenedness over this time period for each time it is shoook.